### PR TITLE
[istio][cni-cilium] Adding the ingressClass parameter in moduleConfigs

### PR DESCRIPTION
**Pull request by AndreyPavlovFlant**
## Description

Adding the ingressClass parameter in moduleConfigs

## Why do we need it, and what problem does it solve?

Currently, the cilium (hubble) and istio (kiali, api-proxy, and metadata-exporter) components can only be published from the default (global) ingress-controller. There is no option to move them to separate nodes.

## What is the expected result?

The client can publish resources from the dedicated ingress-controllers (nodes)

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: istio, cni-cilium
type: chore
summary: Adding the ingressClass parameter in moduleConfigs
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/aws'...

File: modules/110-istio/openapi/config-values.yaml [hunk 1]
```diff
@@ -24,6 +24,13 @@ properties:
       - `AllowAny` — allow.
       - `RegistryOnly` — deny. In this case to work with external services you need to register them with ServiceEntry custom resource or to organize egressgateway.
     default: AllowAny
+  ingressClass:
+    type: string
+    pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
+    description: |
+      The class of the Ingress controller used for the Istio.
```
Comment (apolovov) [inline modules/110-istio/openapi/config-values.yaml:31]:
```suggestion
      The class of the Ingress controller used for Kiali, metadata-exporter and proxy-api.
```

File: modules/110-istio/openapi/config-values.yaml [hunk 2]
```diff
@@ -24,6 +24,13 @@ properties:
...
+
+      Optional. By default, the `modules.ingressClass` global value is used
```
Comment (apolovov) [inline modules/110-istio/openapi/config-values.yaml:33]:
```suggestion
      Optional. By default, the `modules.ingressClass` global value is used.
```

File: modules/110-istio/openapi/doc-ru-config-values.yaml [hunk 3]
```diff
@@ -12,6 +12,11 @@ properties:
       Как поступать с запросами в сторону внешних, то есть не учтенных в mesh'е сервисов:
       - `AllowAny` — пропускать;
       - `RegistryOnly` — запрещать. В данном случае для работы с внешними сервисами потребуется регистрировать их с помощью ресурса ServiceEntry или организовывать egressgateway.
+  ingressClass:
+    description: |
+      Класс Ingress-контроллера, который используется для Istio.
+
+      Опциональный параметр, по умолчанию используется глобальное значение `modules.ingressClass`
```
Comment (apolovov) [inline modules/110-istio/openapi/doc-ru-config-values.yaml:19]:
```suggestion
      Класс Ingress-контроллера, который используется для Kiali, metadata-exporter и api-proxy.

      Опциональный параметр, по умолчанию используется глобальное значение `modules.ingressClass`.
```

File: modules/500-cilium-hubble/openapi/config-values.yaml [hunk 4]
```diff
@@ -5,6 +5,13 @@ properties:
     type: boolean
     default: false
     description: Enabled debug logging for Cilium Hubble component.
+  ingressClass:
+    type: string
+    pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
+    description: |
+      The class of the Ingress controller used for the Cilium Hubble component.
+
+      Optional. By default, the `modules.ingressClass` global value is used
```
Comment (apolovov) [inline modules/500-cilium-hubble/openapi/config-values.yaml:14]:
```suggestion
      Optional. By default, the `modules.ingressClass` global value is used.
```

File: modules/500-cilium-hubble/openapi/doc-ru-config-values.yaml [hunk 5]
```diff
@@ -2,6 +2,11 @@ type: object
 properties:
   debugLogging:
     description: Включает отладочный уровень логирования для компонентов Hubble.
+  ingressClass:
+    description: |
+      Класс Ingress-контроллера, который используется для компонентов Hubble.
+
+      Опциональный параметр, по умолчанию используется глобальное значение `modules.ingressClass`
```
Comment (apolovov) [inline modules/500-cilium-hubble/openapi/doc-ru-config-values.yaml:9]:
```suggestion
      Класс Ingress-контроллера, который используется для Hubble.

      Опциональный параметр, по умолчанию используется глобальное значение `modules.ingressClass`.
```

File: modules/500-cilium-hubble/openapi/config-values.yaml [hunk 6]
```diff
@@ -5,6 +5,13 @@ properties:
     type: boolean
     default: false
     description: Enabled debug logging for Cilium Hubble component.
+  ingressClass:
+    type: string
+    pattern: '^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$'
+    description: |
+      The class of the Ingress controller used for the Cilium Hubble component.
```
Comment (apolovov) [inline modules/500-cilium-hubble/openapi/config-values.yaml:12]:
```suggestion
      The class of the Ingress controller used for Hubble.
```

File: modules/500-cilium-hubble/openapi/doc-ru-config-values.yaml [hunk 7]
```diff
@@ -2,6 +2,11 @@ type: object
 properties:
   debugLogging:
     description: Включает отладочный уровень логирования для компонентов Hubble.
+  ingressClass:
+    description: |
+      Класс Ingress-контроллера, который используется для компонентов Hubble.
```
Comment (apolovov) [inline modules/500-cilium-hubble/openapi/doc-ru-config-values.yaml:7]:
```suggestion
      Класс Ingress-контроллера, который используется для Hubble.
```

**Review by apolovov — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @AndreyPavlovFlant. I've started the workflow for label 'e2e/run/aws'...

**Review by nabokihms — APPROVED** [review]

**Pull request merged**